doc/sphinx/user-extensions/syntax-extensions.rst

```diff
@@ -215,6 +215,8 @@ definition is the following:
 More generally, it is required that notations are explicitly factorized on the
 left. See the next section for more about factorization.
 
+.. _NotationFactorization:
+
 Simple factorization rules
```

Comment:
This section doesn't explain why the user cares about factorization, so hard to follow.  Perhaps you could add a sentence or two to improve this? Maybe something like:
- distinct notations must be compatible
- in some cases, incompatible notations can be supported by factorizing them.
- "factorizing" means <add short description>  (it's a generic term but we have a very specific meaning in mind, described in the section)

IIUC the point is that if one Notation is a prefix of another, you need to factorize the common prefix so that both can be parsed correctly.